python/lsst/ts/watcher/alarm.py

```diff
@@ -0,0 +1,205 @@
+# This file is part of ts_watcher.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["Alarm"]
+
+import time
+
+from lsst.ts.idl.enums.Watcher import AlarmSeverity
+from lsst.ts import salobj
+
+
+class Alarm:
+    """A Watcher alarm.
+
+    Parameters
+    ----------
+    name : `str`
+        Name of alarm. This must be unique among all alarms
+        and should be of the form system_[subsystem_...]_name
+        so that groups of related alarms can be acknowledged.
+        Use underscores instead of dots to make the regular expressions
+        or glob expressions easier to write.
+    callback : ``callable``
+        Function or coroutine to call whenever the alarm changes state.
+    """
+    def __init__(self, name, callback):
+        self.name = name
+        self.callback = callback
+
+        self.severity = AlarmSeverity.NONE
+        self.max_severity = AlarmSeverity.NONE
+        self.reason = ""
+        self.acknowledged = False
+        self.acknowledged_by = ""
+        self.enabled = True
+
+        self.timestamp_severity_oldest = 0
+        self.timestamp_severity_newest = 0
+        self.timestamp_max_severity = 0
+        self.timestamp_acknowledged = 0
+
+    def __eq__(self, other):
```

Comment:
I think I read in the XML (Events - line 32) that you say that each alarm should have a unique name - therefore an Alarm object would only have to match on name to be the same object - isn't that correct?   Or am I missing some nuance here...

Reply:
The equality test is primarily intended for unit testing. In many tests I call a method that is expected to change or not change the state on an Alarm. In such cases I want to be able to test this by checking equality against a copy.

Reply:
Could we put a comment here for that - one could almost argue it belongs in the testing class since it is testing only - but you can go either way about this.